Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Sensor.h

```diff
@@ -106,10 +113,70 @@ class CARLA_API ASensor : public AActor
   ///
   /// You need to provide a reference to self, this is necessary for template
   /// deduction.
-  template <typename SensorT>
-  FAsyncDataStream GetDataStream(const SensorT &Self)
+  template <typename SensorType>
+  FAsyncDataStream GetDataStream(SensorType&& Self)
+  {
+    return Stream.MakeAsyncDataStream<std::remove_cvref_t<SensorType>>(
+      std::forward<SensorType>(Self),
+      GetEpisode().GetElapsedGameTime());
+  }
+
+  template <
+    typename SensorType,
+    typename PixelType>
+  static void SendImageDataToClient(
```

Comment:
Interesting, I think this refactoring may help with the issue of PixelReader bottleneck.